Practice Areas/Engineering/code_review_guidelines.md

```diff
@@ -0,0 +1,37 @@
+# Platform code review guidelines
+
+This document covers rules for engaging the VSP team in [pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews). These guidelines are intended to:
+
+* Help maintain the code quality and stability of the platform itself
+* Enable VSP team to quickly review changes for VFS teams
+
+Before sending your code to a VSP team for review, your code should meet all of the guidelines below.
+
+*Hint*: smaller, more frequent pull requests will make this easier for everyone!
+
+## Guidelines
+
+* **Review lifecycle**
+    * Before sending code to VSP team, have it reviewed by your team (see **First review** below).
+    * An engineer from the VSP willaim to review within 24 hours (business days) and each additional round of changes will be reviewed within 24 hours.  If you need it faster, escalate to Leah and Patrick.
+    * After sending your code for review, don't add additional changes! Code needs to be stable for a safe review.
+    * If non-trivial changes to your code are made *after approval by VSP team*, your change is subject to being reverted.
+
+* **First review**: your code should first be reviewed and approved by a member of your team. That first review should cover:
+    * **Correctness** of the code (does it correctly implement the described feature?)
+    * **Code quality**: is it readable? Is it language-idiomatic?
+    * **Does it have screenshots of the change?** (FE changes)
+
+* **Description**: make it clear what the change is describing.
+    * **PR title** should be a concise summary of the change.
+    * In your **PR description**, discuss how code changes are related to the pull request objectives.
+    * Reference the **GitHub issues** your PR relates to, in order to provide reviewers more context.
+
+* **Size/complexity**: keep your changes scoped to one feature at a time. This means:
+    * **Size**: changes should be smaller than **300** changed lines (not including tests or whitespace). *If your change can't be smaller than this, talk to the team in #vfs-platform-support before asking for a review.*
+    * **One thing at a time**: feature code, cleanup/refactoring changes, dependency/package upgrades, and database migrations should all be in **separate PRs**. As much as possible, each PR should address a single feature or bug at a time.
```

Comment:
:+1: 

Is it worth mentioning minor refactoring and cleanup are okay? I definitely don't want hundreds of lines of cleanup, but if the guidelines bar _any_ cleanup / refactoring without the overhead of a separate PR, that might not be awesome.

Reply:
🤷‍♀ I think this is tricky.  I recently reviewed a PR that had a LOT of good cleanup. Each thing individually was great, but the PR was massive.  

* Rename a variable to be a word and not a single letter? yes please!
* reformat the JSON with a linter? *shrug*
* Fix a one-liner that you missed in your previous PR? no biggie!
* all of the above in a PR that has 80+ comments? NO!!!

Reply:
🤷‍♀ I absolutely agree with your point, and I'm torn.  Cleanup is fine, within reason, but I've seen unreasonable recently too.